epochStart/shardchain/trigger.go

```diff
@@ -226,20 +229,35 @@ func (t *trigger) RequestEpochStartIfNeeded(interceptedHeader data.HeaderHandler
 	}
 }
 
+func (t *trigger) checkIfEpochFinalityAttestingWithLowerRound(metaHdr *block.MetaBlock) {
+	if metaHdr.GetNonce() != t.epochStartNonce+1 {
```

Comment:
This check is done below in IsHeaderConstructionValid method so I think it could be removed from here

Reply:
done